dbt_platform_helper/custom_resources/app_user.py

```diff
@@ -0,0 +1,204 @@
+import json
+import logging
+import time
+from urllib import request
+from urllib.error import HTTPError
+
+import boto3
+import psycopg2
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger(__name__)
+
+
+def drop_user(cursor, username):
+    cursor.execute(f"SELECT * FROM pg_catalog.pg_user WHERE usename = '{username}'")
+
+    if cursor.fetchone() is not None:
+        cursor.execute(f"GRANT {username} TO postgres")
+        cursor.execute(f"DROP OWNED BY {username}")
+        cursor.execute(f"DROP USER {username}")
+
+
+def create_db_user(conn, cursor, username, password, permissions):
+    drop_user(cursor, username)
+
+    cursor.execute(f"CREATE USER {username} WITH ENCRYPTED PASSWORD '%s'" % password)
+    cursor.execute(f"GRANT {username} to postgres;")
+    cursor.execute(f"GRANT {', '.join(permissions)} ON ALL TABLES IN SCHEMA public TO {username};")
+    cursor.execute(
+        f"ALTER DEFAULT PRIVILEGES FOR USER {username} IN SCHEMA public GRANT {', '.join(permissions)} ON TABLES TO {username};"
+    )
+    conn.commit()
+
+
+def create_or_update_user_secret(ssm, user_secret_name, user_secret_string, event):
+    user_secret_description = event["ResourceProperties"]["SecretDescription"]
+    copilot_application = event["ResourceProperties"]["CopilotApplication"]
+    copilot_environment = event["ResourceProperties"]["CopilotEnvironment"]
+
+    user_secret = None
+
+    try:
+        user_secret = ssm.put_parameter(
+            Name=user_secret_name,
+            Description=user_secret_description,
+            Value=json.dumps(user_secret_string),
+            Tags=[
+                {
+                    "Key": "custom:cloudformation:stack-name",
+                    "Value": event["StackId"].split("/")[1],
+                },
+                {"Key": "custom:cloudformation:logical-id", "Value": event["LogicalResourceId"]},
+                {"Key": "custom:cloudformation:stack-id", "Value": event["StackId"]},
+                {"Key": "copilot-application", "Value": copilot_application},
+                {"Key": "copilot-environment", "Value": copilot_environment},
+            ],
+            Type="String",
+        )
+    except ClientError as error:
+        if error.response["Error"]["Code"] == "ParameterAlreadyExists":
+            user_secret = ssm.put_parameter(
+                Name=user_secret_name,
+                Description=user_secret_description,
+                Value=json.dumps(user_secret_string),
+                Overwrite=True,
+            )
+
+    return user_secret
+
+
+def send(event, body, logger, headers):
```

Comment:
This method is new, though closely modelled on logic in `dbt_platform_helper/custom_resources/s3_object.py`. Wasn't able to make it a shared util because it won't import in lambda env